doc/man3/EVP_PKEY_set1_RSA.pod

```diff
@@ -89,6 +93,11 @@ Previous versions of this document suggested using EVP_PKEY_type(pkey->type)
 to determine the type of a key. Since B<EVP_PKEY> is now opaque this
 is no longer possible: the equivalent is EVP_PKEY_base_id(pkey).
 
+Currently, the behaviors of EVP_PKEY_check are not consistent for different
+public key algorithms. For instance, an RSA key will be checked for both
+public and private components, but for EC keys they could be checked
+separately for the public or the private components.
```

Comment:
It could be argued that `EVP_PKEY_check` *should* be consistent and that the `EVP_PKEY_ASN1_METHOD` checking functions must be made to support that consistency.  So that leaves it up to you to decide if `EVP_PKEY_check` should support public keys or not.

Reply:
I'd like to strict this `EVP_PKEY_check` to deal with key-pair only (that means it will fail if a public key is passed), and for public key check, there could be another function like `EVP_PKEY_check_public` or similar.

To stay consistent for `EVP_PKEY_check`, I will remove the support of EC key check in this PR and later I will probably write a new wrapper around `EC_KEY_check_key` to provide key-pair check functionality only for EC keys. Also for DH and DSA keys, similar effort should be done.